lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -102,167 +237,292 @@ def get_subnet_info(subnet):
         subnet = camel_dict_to_snake_dict(subnet)
 
     if 'tags' in subnet:
-        subnet['tags'] = boto3_tag_list_to_ansible_dict(subnet['tags'])
+        subnet['tags'] = boto3_tag_list_to_ansible_dict(subnet['tags'], 'key', 'value')
```

Comment:
You can remove 'key' and 'value' since backwards compatibility was fixed for this in 56fe9499d14fa6e238fd3f32b1b33e6e7524e753.